### PR TITLE
LEXI-162 Create ClayDropdown

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

**Comment by jbalsas** [comment]
@carloslancha, merge conflicts ahead!!

**Comment by carloslancha** [comment]
This component is the base to create Dropdown menu (https://issues.liferay.com/browse/LEXI-162) and Dropdown Action Menu (https://issues.liferay.com/browse/LEXI-124) components

File: packages/clay-dropdown/src/ClayDropdownBase.soy [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+/**
+ * Renders a search input
+ */
+{template .search}
+	{@param spritemap: string}
+
+	<div class="dropdown-section">
+		<div class="input-group input-group-inset">
+			<div class="input-group-input">
+				<input class="form-control" data-oninput="_handleSearch" placeholder="Search for..." type="text" ref="searchInput">
```
Comment (matuzalemsteles) [inline packages/clay-dropdown/src/ClayDropdownBase.soy:92]: Would not it be better to pass the `placeholder` via API? For cases of other languages.

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-dropdown/src/ClayDropdownBase.soy [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+				{param url: $item.url /}
+			{/delcall}
+		{/if}
+	{/foreach}
+{/template}
+
+/**
+ * Renders a search input
+ */
+{template .search}
```
Comment (matuzalemsteles) [inline packages/clay-dropdown/src/ClayDropdownBase.soy:86]: @carloslancha, as we are doing the search by js and letting `caption` to customize the message, dev could not customize in the case of the demo. Maybe emit an `event`?

File: packages/clay-dropdown/src/ClayDropdownBase.soy (hunk 1 above)
Comment (carloslancha) [reply packages/clay-dropdown/src/ClayDropdownBase.soy:92]: We still need to think about how to manage languages like in ClayBadge. I've just opened an issue for that https://github.com/metal/metal-clay-components/issues/95

File: packages/clay-dropdown/src/ClayDropdownBase.soy (hunk 2 above)
Comment (carloslancha) [reply packages/clay-dropdown/src/ClayDropdownBase.soy:86]: Done https://github.com/metal/metal-clay-components/pull/93/files#diff-1dace496b1f2a92f6bf7984f5a1949a1R41 👍

**Comment by jbalsas** [comment]
Awesome, @carloslancha!!

@matuzalemsteles, can you take a look to see if there's anything you'd like to comment before merging?

File: packages/clay-dropdown/src/ClayDropdownBase.soy (hunk 2 above)
Comment (matuzalemsteles) [reply packages/clay-dropdown/src/ClayDropdownBase.soy:86]: Thanks @carloslancha!

**Comment by matuzalemsteles** [comment]
Awesome!!

@jbalsas LGTM 🎉!

**Pull request merged**